### ISSUE TITLE
BlankNodes getting converted literals

### ISSUE THREAD

**Issue by jeswr** [eyereasoner/eye#76]
For the following  `data.n3`
```
{ _:n3-30 ?p ?o} => {} .
```

the following is output
```
$ npx eyereasoner@latest --nope --quiet --pass-all ./data.n3 
{"some1-301" ?U_0 ?U_2} => true.
```

The problem seems to specifically be when there is a dash (`-`) in the blank node similar problems occur if there is a `.` in the blank node name.

**Comment by jeswr** [comment]
In lieu of a good production grammar for N3, I would expect BlankNodes to be parsed in the same way as in turtle, which is given by the following https://www.w3.org/TR/turtle/#grammar-production-BlankNode.

**Comment by jeswr** [comment]
I would also expect an error to be thrown if it *was* a literal because as far as I am aware literals are not valid subjects in N3?

**Comment by josd** [comment]
Is for sure a bug and a side effect of the internal bookkeeping of the whole zoo of variables.

**Comment by josd** [comment]
> I would also expect an error to be thrown if it _was_ a literal because as far as I am aware literals are not valid subjects in N3?

Literals are valid subjects in N3
https://w3c.github.io/N3/spec/#grammar-production-subject
https://w3c.github.io/N3/spec/#grammar-production-expression
https://w3c.github.io/N3/spec/#grammar-production-path
https://w3c.github.io/N3/spec/#grammar-production-pathItem

**Comment by josd** [comment]
It is now fixed in EYE v3.4.3 and giving
```
{?U_1-301 ?U_0 ?U_2} => true.
```
Remark that an existential within the scope of the premis can become a universal within the scope of the rule.
